pkg/scale/decode.go

```diff
@@ -121,46 +119,94 @@ func (ds *decodeState) unmarshal(dstv reflect.Value) (err error) {
 	switch in.(type) {
 	case *big.Int:
 		err = ds.decodeBigInt(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding big integer: %w", err)
+		}
 	case *Uint128:
 		err = ds.decodeUint128(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding Uint128: %w", err)
+		}
 	case int, uint:
 		err = ds.decodeUint(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding uint: %w", err)
+		}
 	case int8, uint8, int16, uint16, int32, uint32, int64, uint64:
 		err = ds.decodeFixedWidthInt(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding fixed width int: %w", err)
+		}
 	case []byte:
 		err = ds.decodeBytes(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding []byte: %w", err)
+		}
 	case string:
 		err = ds.decodeBytes(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding string: %w", err)
+		}
 	case bool:
 		err = ds.decodeBool(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding bool: %w", err)
+		}
 	case Result:
 		err = ds.decodeResult(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding Result: %w", err)
+		}
 	case VaryingDataType:
 		err = ds.decodeVaryingDataType(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding VaryingDataType: %w", err)
+		}
 	case VaryingDataTypeSlice:
 		err = ds.decodeVaryingDataTypeSlice(dstv)
+		if err != nil {
+			err = fmt.Errorf("decoding VaryingDataTypeSlice: %w", err)
+		}
 	default:
 		t := reflect.TypeOf(in)
 		switch t.Kind() {
 		case reflect.Bool, reflect.Int, reflect.Int8, reflect.Int16,
 			reflect.Int32, reflect.Int64, reflect.String, reflect.Uint,
 			reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
 			err = ds.decodeCustomPrimitive(dstv)
+			if err != nil {
+				err = fmt.Errorf("decoding Custom Primitive: %w", err)
```

Comment:
nit I don't think we need to use capitalized words, since we don't need camelCase for error string messages.

```suggestion
				err = fmt.Errorf("decoding custom primitive: %w", err)
```